### PR TITLE
refactor: create helper methods to simplify generator code

### PR DESCRIPTION
**Pull request by beeman**

### CODE REVIEW

File: lib/inputs.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { valueRequired } = require('./validators')
+
+const input = (name, message, required = false) => ({
```
Comment (beeman) [inline lib/inputs.js:3]: A simple wrapper around the input types we use. Similar to FormService

File: lib/validators.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+const valueRequired = (value, name) => /.+/.test(value) ? true : `Input ${name} is required`
```
Comment (beeman) [inline lib/validators.js:2]: Extract validation methods into a separate file

File: templates/admin-module.js [hunk 3]
```diff
@@ -1,78 +1,81 @@
 const chalk = require('chalk')
+const path = require('path')
 const log = require('../lib/logger')
 
+const {input, checkbox} = require('../lib/inputs')
+
+const templatePath = 'templates/admin-module'
```
Comment (beeman) [inline templates/admin-module.js:7]: This is a constant for this template

File: templates/admin-module.js [hunk 4]
```diff
@@ -1,78 +1,81 @@
...
-  },
-}, {
-  type: 'checkbox',
-  name: 'moduleItems',
-  message: 'Select the items you want to include:',
-  choices: [
+const prompts = [
+  input('moduleName', 'Module Name:', true),
+  input('modelName', 'Primary Name:', true),
+  checkbox('moduleItems', 'Select the items you want to include:', [
```
Comment (beeman) [inline templates/admin-module.js:13]: Having these prompts extracted saves a lot of duplication

File: templates/admin-module.js [hunk 5]
```diff
@@ -1,78 +1,81 @@
...
+  const moduleName = '{{ dashCase moduleName }}'
+
+  const actions = []
+
+  // Push and add action to the actions array, prepend target and template path
+  const addFile = (templateFile, ...targetFiles) => actions.push({
+    type: 'add',
+    templateFile: path.join(templatePath, templateFile),
+    path: path.join(targetPath, ...targetFiles),
+  })
```
Comment (beeman) [inline templates/admin-module.js:33]: Pulling out the addFile method and prefixing the paths saves a lot of duplication.

File: templates/admin-module.js [hunk 6]
```diff
@@ -1,78 +1,81 @@
...
-      type: 'add',
-      path: '{{ lowerCase moduleName }}/{{ dashCase moduleName }}.routes.ts',
-      templateFile: 'templates/admin-module/routes.txt',
-    })
-  }
+  // Loop over all the selected moduleItems and execute their handler.
+  data.moduleItems.forEach(moduleItem => handlers[moduleItem]
+    ? handlers[moduleItem]()
+    : log.red.b(`Handler for ${moduleItem} not defined.`)
+  )
```
Comment (beeman) [inline templates/admin-module.js:78]: Loop through all the selected items and execute their handler if it's found.

File: templates/admin-module.js [hunk 7]
```diff
@@ -1,78 +1,81 @@
...
-    log.white('Adding Service...')
-    actions.push({
-      type: 'add',
-      path: '{{ lowerCase moduleName }}/{{ dashCase moduleName }}.service.ts',
-      templateFile: 'templates/admin-module/service.txt',
-    })
+      containers.forEach(component => {
+        addFile(`containers/${component}.txt`, `containers/${moduleName}.${component}.ts`)
+      })
+    }
```
Comment (beeman) [inline templates/admin-module.js:68]: This are all the handlers that can be executed.

**Review by brannon-darby — APPROVED** [review]
lgtm

**Pull request merged**